inc/computeroperatingsystem.class.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/*
+ * @version $Id$
+ -------------------------------------------------------------------------
+ GLPI - Gestionnaire Libre de Parc Informatique
+ Copyright (C) 2016 Teclib'.
+
+ http://glpi-project.org
+
+ based on GLPI - Gestionnaire Libre de Parc Informatique
+ Copyright (C) 2003-2014 by the INDEPNET Development Team.
+
+ -------------------------------------------------------------------------
+
+ LICENSE
+
+ This file is part of GLPI.
+
+ GLPI is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+
+ GLPI is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ --------------------------------------------------------------------------
+ */
+
+/** @file
+* @brief Operating system management
+*/
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * ComputerOperatingSystem Class
+ *
+ * Class to manage operating systems
+**/
+class ComputerOperatingSystem extends CommonDBChild {
+
+   // From CommonDBChild
+   static public $itemtype = 'Computer';
+   public $dohistory       = true;
+
+   public function __construct()
+   {
+       parent::__construct();
+       $this->forceTable('glpi_computers');
+   }
+
+   static function getTypeName($nb=0) {
+      return _n('Operating system', 'Operating systems', $nb);
+   }
+
+
+   /**
+    * @see CommonGLPI::getTabNameForItem()
+   **/
+   function getTabNameForItem(CommonGLPI $item, $withtemplate=0) {
+
+      if (!$withtemplate
+          && ($item->getType() == 'Computer')
+          && Computer::canView()) {
+         $nb = 0;
+         return self::createTabEntry(self::getTypeName(), $nb);
+      }
+      return '';
+   }
+
+
+   /**
+    * @see CommonGLPI::defineTabs()
+    *
+    * @since version 0.85
+   **/
+   function defineTabs($options=array()) {
+
+      $ong = array();
+      $this->addDefaultFormTab($ong);
+
+      return $ong;
+   }
+
+
+   /**
+    * @param $item         CommonGLPI object
+    * @param $tabnum       (default 1)
+    * @param $withtemplate (default 0)
+   **/
+   static function displayTabContentForItem(CommonGLPI $item, $tabnum=1, $withtemplate=0) {
+
+      self::showForComputer($item);
+      return true;
+   }
+
+
+   /**
+    * Print the computer's operating system form
+    *
+    * @param $comp Computer object
+    *
+    * @return Nothing (call to classes members)
+   **/
+   static function showForComputer(Computer $comp) {
+      global $DB;
+
+      if (!Session::haveRight("computer", UPDATE)) {
```

Comment:
Should be $comp->can() to check this specific computer right (not the type right)